### PR TITLE
[net8.0] Merge main to net8.0

### PR DESCRIPTION
**Pull request by rmarinho**
### Description of Change

Bring latest changes from main to net8 branch.
Fixes UITests

### CODE REVIEW

**Review by mattleibow — DISMISSED** [review]

**Comment by rmarinho** [comment]
@Eilon can you investigate why Blazor tests are failing?

**Comment by rmarinho** [comment]
@jfversluis This bring more windows tests to net8, can you double check is ok ?

**Comment by Eilon** [comment]
@rmarinho - at least one of the failures I see for BlazorWebView isn't the tests failing, but rather not even compiling:

```
/Users/builder/azdo/_work/1/s/src/BlazorWebView/tests/MauiDeviceTests/Elements/BlazorWebViewTests.cs(144,5): error RS0030: The symbol 'ServiceCollectionServiceExtensions.AddScoped<TService>(IServiceCollection)' is banned in this project: Use a Factory method to register the service instead (https://github.com/dotnet/roslyn-analyzers/blob/main/src/Microsoft.CodeAnalysis.BannedApiAnalyzers/BannedApiAnalyzers.Help.md) [/Users/builder/azdo/_work/1/s/src/BlazorWebView/tests/MauiDeviceTests/MauiBlazorWebView.DeviceTests.csproj::TargetFramework=net8.0-ios]
/Users/builder/azdo/_work/1/s/src/BlazorWebView/tests/MauiDeviceTests/Elements/BlazorWebViewTests.cs(186,5): error RS0030: The symbol 'ServiceCollectionServiceExtensions.AddScoped<TService>(IServiceCollection)' is banned in this project: Use a Factory method to register the service instead (https://github.com/dotnet/roslyn-analyzers/blob/main/src/Microsoft.CodeAnalysis.BannedApiAnalyzers/BannedApiAnalyzers.Help.md) [/Users/builder/azdo/_work/1/s/src/BlazorWebView/tests/MauiDeviceTests/MauiBlazorWebView.DeviceTests.csproj::TargetFramework=net8.0-ios]
/Users/builder/azdo/_work/1/s/src/BlazorWebView/tests/MauiDeviceTests/Elements/BlazorWebViewTests.cs(214,5): error RS0030: The symbol 'ServiceCollectionServiceExtensions.AddScoped<TService>(IServiceCollection)' is banned in this project: Use a Factory method to register the service instead (https://github.com/dotnet/roslyn-analyzers/blob/main/src/Microsoft.CodeAnalysis.BannedApiAnalyzers/BannedApiAnalyzers.Help.md) [/Users/builder/azdo/_work/1/s/src/BlazorWebView/tests/MauiDeviceTests/MauiBlazorWebView.DeviceTests.csproj::TargetFramework=net8.0-ios]
```

@jonathanpeppers - are these symbols supposed to be banned in tests or just prod code? Or maybe that was changed in another PR you sent?

File: src/Controls/tests/UITests/snapshots/windows/Issue15330Test.png
Comment (jfversluis) [inline src/Controls/tests/UITests/snapshots/windows/Issue15330Test.png:1]: I don't think this should have been committed @PureWeen ?

Comment (jfversluis) [reply src/Controls/tests/UITests/snapshots/windows/Issue15330Test.png:1]: There is another one just below

**Comment by jfversluis** [comment]
@rmarinho yeah the tests aren't even running yet as Eilon mentions. Tests seem OK to me from a glance. But it's the banned symbols thing that needs to be resolved first.

**Comment by jfversluis** [comment]
@Eilon you approved it yourself 😬  https://github.com/dotnet/maui/pull/17004

Weird that it didn't give any errors before. So I guess the calls for the BlazorWebView tests need some updating

**Comment by jonathanpeppers** [comment]
Yes, `$(IsTestProject)` was added in https://github.com/dotnet/maui/commit/e7812b0f00ffebfe753cc8f35479aa7a4fbf6135 if you need to turn off the BannedApisAnalyzer in a test project.

Generally, it should be fine to use these "banned apis" in tests, as they are usually banned for performance reasons.

**Comment by Eilon** [comment]
All green now.

**Review by Eilon — APPROVED** [review]

**Review by PureWeen — APPROVED** [review]

**Pull request merged**